modules/ROOT/pages/client-id-based-policies.adoc

```diff
@@ -14,48 +14,73 @@ endif::[]
+The Client ID Enforcement Policy restricts access to a protected resource, by only allowing HTTP requests if credentials of client applications already xref:api-contracts-landing-page.adoc[registered to (contracted with) the API being targeted] are provided in each request. In Anypoint Platform, the client ID and client secret pair is also known as credentials.
 
-The policy validates the Client ID and, optionally, the Client Secret, of a client application created within an Anypoint Platform Organization.
+After the Client ID Enforcement policy is applied, the Mule runtime engine (Mule) caches the information locally and updates it periodically. In addition to performance improvements, caching enables the policy to work offline during connectivity outages. 
 
-IMPORTANT: This policy will not validate successfully any client application credentials generated outside of the Anypoint Platform context.
+Regardless of the state of the management plane, the credentials are enforced with the existing data. Once the connectivity is re-established, the cache resumes periodical updates.
 
-NOTE: The set of Client ID and Client Secret are also known as credentials.
+//What information is stored or retrieved. -- Eva to send
+//Latest version has encryption. So the information is encrypted.
 
-= Analytics and  access tracking
-If you need to track client application access to your API, apply a policy that directly or indirectly makes sure that client ID is provided in the HTTP request (ie Client ID enforcement type policy).
+// How does the policy work?
+//Where can you get the inforamtion for the policy--Mariano to send me the information
 
-= Indirect client ID enforcement
-There are other provided policies that enforce client application credentials. For example:
 
-* xref:rate-limiting-and-throttling-sla-based-policies.adoc[Rate Limiting - SLA-Based Policies]
+Default policies that internally enforce client application credentials include:
 
+* xref:rate-limiting-and-throttling-sla-based-policies.adoc[Rate Limiting - SLA-Based Policies]
++
 In Mule 3.8.5 or later, the following OAuth 2.0 Token Enforcement policies include a client ID enforcement validation:
-
 * xref:external-oauth-2.0-token-validation-policy.adoc[Mule OAuth 2.0 Access Token Enforcement]
 * xref:policy-ping-federate.adoc[PingFederate OAuth 2.0 Token Enforcement]
 * xref:policy-openid-connect.adoc[OpenID Connect OAuth 2.0 Token Enforcement]
 
-= Offline mode
+== Policy Prerequisites and Restrictions
 
-Credentials data from client application credentials registered to protected APIs, are provided by Anypoint Platform. Once a policy requiring such information is applied (eg Client ID enforcement policy), the Runtime caches persistently the information locally updating it periodically.
+For the policy to work successfully, you must:
```

Comment:
I don't think there is any prerequisite for the policy or restrictions. But if you want to have a section of "how the policy works":

Every client application that wants to consume an API protected by a Client ID enforcement policy needs to previously have requested access to this API. Once there is an approved contract between the application and the API, every request needs to include the client credentials in compliance with how the policy has been configured. 

As an example, if the policy has been configured to expect a client id and client secret as headers, the application needs to make sure the credentials are sent in the request using the corresponding headers. 

I would also have a different section for "API Specification" that includes the rest console information:

The client id enforcement policy will require changes in your API specification to reflect the need to provide the credentials. 
You can find the RAML/OAS snippet link containing the RAML/OAS code you need to add to the API spec in the corresponding policy from the list of applied policies in the Policies tab of your API specification in API Manager.

Reply:
Done! Thanks!